### ISSUE TITLE
Use horovod to istributed training how to deal with batch_normalize layer in model?

### ISSUE THREAD

**Issue by yjiangling** [horovod/horovod#2808]
There are some batch_normalize layers in my model, if I want to train the model with multi GPUs with horovod, should I use the following codes?
**code1:**

with tf.control_dependencies(tf.get_collection(tf.GraphKeys.UPDATE_OPS)):
	

> optimizer = tf.train.AdamOptimizer(lr, beta1=0.9, beta2=0.98, epsilon=1e-09)

optimizer = hvd.DistributedOptimizer(optimizer)
train_op = optimizer.minimize(loss, global_step=global_step)

or should I write it like this?
**code2:**

with tf.control_dependencies(tf.get_collection(tf.GraphKeys.UPDATE_OPS)):
	

> optimizer = tf.train.AdamOptimizer(lr, beta1=0.9, beta2=0.98, epsilon=1e-09)

        

> optimizer = hvd.DistributedOptimizer(optimizer)

train_op = optimizer.minimize(loss, global_step=global_step)

Anyone can give me some help? Thans a lot.

**Comment by chongxiaoc** [comment]
Can you check the `examples/tensorflow` and `examples/tensorflow2`?

**Comment by yjiangling** [comment]
> Can you check the `examples/tensorflow` and `examples/tensorflow2`?

OK, THANKS  A LOT.
